packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -27,80 +27,76 @@ const sanitizeOverFlowingChildren = (spanMap: Map<string, Span>): Map<string, Sp
+      return;
+    }
+    // parentSpan will be undefined when its parentSpan is dropped previously
+    const parentSpan = spanMap.get(span.references[0].spanID);
 
-          case span.startTime >= parentEndTime || childEndTime <= parentSpan.startTime:
-            // case 5: child outside of parent range => drop the child span
-            //      |----parent----|
-            //                        |----child--|
-            // or
-            //                      |----parent----|
-            //       |----child--|
+    if (!parentSpan) {
+      // Drop the child spans of dropped parent span
+      spanMap.delete(span.spanID);
+      return;
+    }
+    const childEndTime = span.startTime + span.duration;
+    const parentEndTime = parentSpan.startTime + parentSpan.duration;
+    if (span.startTime >= parentSpan.startTime) {
+      if (span.startTime >= parentEndTime) {
+        // child outside of parent range => drop the child span
+        //      |----parent----|
+        //                        |----child--|
+        // Remove the childSpan from spanMap
+        spanMap.delete(span.spanID);
 
-            // Remove the childSpan from spanMap
-            spanMap.delete(span.spanID);
+        // Remove the childSpanId from its parent span
+        parentSpan.childSpanIds = parentSpan.childSpanIds.filter(id => id !== span.spanID);
+        spanMap.set(parentSpan.spanID, { ...parentSpan });
+        return;
+      }
+      if (childEndTime > parentEndTime) {
+        // child end after parent, truncate is needed
+        //      |----parent----|
+        //              |----child--|
+        spanMap.set(span.spanID, {
+          ...span,
+          duration: parentEndTime - span.startTime,
+        });
+        return;
+      }
+      // everything looks good
+      // |----parent----|
+      //   |----child--|
+      return;
+    }
+    if (childEndTime <= parentSpan.startTime) {
+      // child outside of parent range => drop the child span
+      //                      |----parent----|
+      //       |----child--|
 
-            // Remove the childSpanId from its parent span
-            parentSpan.childSpanIds = parentSpan.childSpanIds.filter(id => id === span.spanID);
-            spanMap.set(parentSpan.spanID, { ...parentSpan });
-            break;
+      // Remove the childSpan from spanMap
+      spanMap.delete(span.spanID);
 
-          default:
-            // Never reaches to default
-            // Something unexpected happened
-            throw RangeError(`Error while computing Critical Path Algorithm.`);
-        }
-      } else {
-        // Drop the child spans of dropped parent span
-        spanMap.delete(span.spanID);
-      }
+      // Remove the childSpanId from its parent span
+      parentSpan.childSpanIds = parentSpan.childSpanIds.filter(id => id !== span.spanID);
+      spanMap.set(parentSpan.spanID, { ...parentSpan });
+    } else if (childEndTime <= parentEndTime) {
```

Comment:
codecov says this branch (child before parent) is not covered by unit tests, is that correct?

Reply:
Yeah, I will to add a new test case. Previously (child before parent )and (child after parent) are in same case block so that both cases are covered as there is already a test case written for (child after parent)

Reply:
Added a new test case !!